lib/simulator/simulatormodule.c

```diff
@@ -952,7 +952,25 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
     gstate = TAKE_GIL();
 
     pSihHdl = PyTuple_GetItem(args, 0);
-    hdl = (gpi_sim_hdl)PyLong_AsUnsignedLong(pSihHdl);
```

Comment:
The value on this case does not come from python though. It is a c pointer
that is placed in a python int then read back out.

I think there is a case to move the coffee that does these operation into
the python version specific code though and use the native ptr types on 3.x
and move to unsigned long long for 2.7 with overflow checking as well.

Stu
On 18 Jun 2016 14:37, "Andreas Hindborg" <notifications@github.com> wrote:

> In lib/simulator/simulatormodule.c
> <https://github.com/potentialventures/cocotb/pull/443#discussion_r67601159>
> :
>
> > @@ -952,7 +952,25 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
> >      gstate = TAKE_GIL();
> >
> >      pSihHdl = PyTuple_GetItem(args, 0);
> > -    hdl = (gpi_sim_hdl)PyLong_AsUnsignedLong(pSihHdl);
>
> Regarding python and long, Python PyLong objects are not guaranteed to be
> 4/8 bytes for 32/64 bit machins. They are arbitrary size integers. From a
> python script you can put a value larger than 2^64 in a PyLong and the
> PyLong_AsUnsigned will return -1 and raise an error when called.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/potentialventures/cocotb/pull/443/files/01de091c8c39d81cee2a2c26ccccfc861f5f1033#r67601159>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADhwy7PSb3Dv_4Hjjqbn1Zy7hPGJ4S98ks5qM_SsgaJpZM4IoZnu>
> .
>